pkg/util/podprobemarker/pod_probe_marker_test.go

```diff
@@ -0,0 +1,503 @@
+/*
+Copyright 2025 The Kruise Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package podprobemarker
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	appsalphav1 "github.com/openkruise/kruise/apis/apps/v1alpha1"
+	"github.com/openkruise/kruise/pkg/util"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func init() {
+	scheme = runtime.NewScheme()
+	utilruntime.Must(corev1.AddToScheme(scheme))
+	utilruntime.Must(appsalphav1.AddToScheme(scheme))
+}
+
+var (
+	scheme *runtime.Scheme
+)
+
+func TestGetPodProbeMarkerForPod(t *testing.T) {
+
+	cases := []struct {
+		name      string
+		ppmList   *appsalphav1.PodProbeMarkerList
+		pod       *corev1.Pod
+		expect    []*appsalphav1.PodProbeMarker
+		expectErr error
+	}{
+		{
+			name: "get pod probe marker list resources",
```

Comment:
add a case for pod with kruise.io/podprobemarker-list annoation

Reply:
202 line already contains kruise.io/podprobemarker-list annoation